src/cogent3/core/location.py

```diff
@@ -1596,6 +1591,202 @@ def make_seq_feature_map(self, align_feature_map: "FeatureMap") -> "FeatureMap":
 
         return FeatureMap(spans=spans, parent_length=self.parent_length)
 
+    @functools.singledispatchmethod
+    def shared_gaps(self, other) -> numpy.ndarray:
+        """returns a numpy array of the shared [(gap start, gap end), ...]
+
+        Notes
+        -----
+        The result is in alignment coordinates
+        """
+        if len(self) != len(other):
+            raise AssertionError(
+                f"{len(other)=} != {len(self)=}, from a different alignment?"
+            )
+        if self.num_gaps == 0 or other.num_gaps == 0:
+            return numpy.array([], dtype=_DEFAULT_GAP_DTYPE)
+
+        return self.shared_gaps(other.get_gap_align_coordinates())
+
+    @shared_gaps.register
+    def _(self, other: numpy.ndarray) -> numpy.ndarray:
+        """returns a numpy array of the shared [(gap start, gap end), ...]
+
+        Notes
+        -----
+        The result is in alignment coordinates
+        """
+        other_gaps = other
+        if not len(other_gaps):
+            return numpy.array([], dtype=_DEFAULT_GAP_DTYPE)
+
+        if other_gaps[-1][-1] > len(self):
+            raise AssertionError(
+                f"other_gaps {other_gaps!r} from a different alignment?"
+            )
+        self_gaps = self.get_gap_align_coordinates()
+
+        result = coords_intersect(self_gaps, other_gaps)
+        return numpy.array(result, dtype=_DEFAULT_GAP_DTYPE)
+
+    @functools.singledispatchmethod
+    def minus_gaps(self, other_gaps):
```

Comment:
`def minus_gaps(self, other_gaps: Union["IndelMap", numpy.ndarray ): -> "IndelMap"`

Reply:
There is a limitation with python3.9, it raises:
```
NameError: name 'IndelMap' is not defined
```